scripts/download-generate-and-diff-latest-schema-types.ts

```diff
@@ -0,0 +1,132 @@
+import fs from "fs";
+import https from "https";
+import os from "os";
+import path from "path";
+import { generate } from "@graphql-codegen/cli";
+import {
+  getConfig,
+  generatedFileName as localGeneratedTypesFileName,
+} from "../codegen";
+
+const GITHUB_API = "https://api.github.com";
+const INITIAL_PATH = "graphql/schema";
+const REPO = "/repos/evergreen-ci/evergreen/contents/";
+const USER_AGENT = "Mozilla/5.0";
+
+const downloadFile = (url: string, savePath: string) =>
+  new Promise((resolve, reject) => {
+    https
+      .get(url, (res) => {
+        const writeStream = fs.createWriteStream(savePath);
+        res.pipe(writeStream);
+
+        writeStream.on("finish", () => {
+          writeStream.close();
+          resolve(true);
+        });
+
+        writeStream.on("error", reject);
+      })
+      .on("error", reject);
+  });
+
+const fetchAndDownloadFiles = (repoPath: string, localPath: string) =>
```

Comment:
[nit] This feels slightly redundant, `fetchFiles` or `downloadFiles` could probably work.